src/Microsoft.ML.TorchSharp/AutoFormerV2/Anchors.cs

```diff
@@ -0,0 +1,153 @@
+    /// </summary>
+    public class Anchors : Module<Tensor, Tensor>
+    {
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "MSML_PrivateFieldName:private field names not in _camelCase format", Justification = "Need to match TorchSharp.")]
+        private readonly int[] pyramidLevels;
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "MSML_PrivateFieldName:private field names not in _camelCase format", Justification = "Need to match TorchSharp.")]
+        private readonly int[] strides;
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "MSML_PrivateFieldName:private field names not in _camelCase format", Justification = "Need to match TorchSharp.")]
+        private readonly int[] sizes;
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "MSML_PrivateFieldName:private field names not in _camelCase format", Justification = "Need to match TorchSharp.")]
+        private readonly double[] ratios;
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "MSML_PrivateFieldName:private field names not in _camelCase format", Justification = "Need to match TorchSharp.")]
+        private readonly double[] scales;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Anchors"/> class.
+        /// </summary>
+        /// <param name="pyramidLevels">Pyramid levels.</param>
+        /// <param name="strides">Strides between adjacent bboxes.</param>
+        /// <param name="sizes">Different sizes for bboxes.</param>
+        /// <param name="ratios">Different ratios for height/width.</param>
+        /// <param name="scales">Scale size of bboxes.</param>
+        public Anchors(int[] pyramidLevels = null, int[] strides = null, int[] sizes = null, double[] ratios = null, double[] scales = null)
+            : base(nameof(Anchors))
+        {
+            this.pyramidLevels = pyramidLevels != null ? pyramidLevels : new int[] { 3, 4, 5, 6, 7 };
+            this.strides = strides != null ? strides : this.pyramidLevels.Select(x => (int)Math.Pow(2, x)).ToArray();
+            this.sizes = sizes != null ? sizes : this.pyramidLevels.Select(x => (int)Math.Pow(2, x + 2)).ToArray();
+            this.ratios = ratios != null ? ratios : new double[] { 0.5, 1, 2 };
+            this.scales = scales != null ? scales : new double[] { Math.Pow(2, 0), Math.Pow(2, 1.0 / 3.0), Math.Pow(2, 2.0 / 3.0) };
+        }
+
+        /// <summary>
+        /// Generate anchors for an image.
+        /// </summary>
+        /// <param name="image">Image in Tensor format.</param>
+        /// <returns>All anchors.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "MSML_GeneralName:This name should be PascalCased", Justification = "Need to match TorchSharp.")]
+        public override Tensor forward(Tensor image)
```

Comment:
Does it need to?

I don't have strong opinions on this but TorchSharp has a pretty good reason to stay feeling like PyTorch/Python. 

IMO - ML.NET should feel like .NET.

Reply:
Again - I don't have strong feelings on this and I haven't traced where this is used. So it might make sense to be like Python.

I just noticed that it's a public method in a public class in ML.NET. So I have an opinion :).

Reply:
Thinking out loud:
Maybe these are what model is being loaded into and names/weights need to match?

Reply:
Talked offline - this is mostly lift and shift of MSR model.